src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -99,17 +115,33 @@ pub struct WorkUnitStore {
 
 #[derive(Default)]
 pub struct WorkUnitInnerStore {
+  graph: WorkunitGraph,
+  span_id_to_graph: HashMap<SpanId, NodeIndex<u32>>,
   workunit_records: HashMap<SpanId, WorkunitRecord>,
   started_ids: Vec<SpanId>,
   completed_ids: Vec<SpanId>,
```

Comment:
It feels like these ids don't really need to be recorded independently... you can get them by scanning the `workunit_records` pretty easily?

Reply:
These are used (along with the `last_seen_` indices) to track what workunits we've already given to the streaming workunit reporter. There might be a better way to do this, but that would be a change for another PR I think.

Reply:
Fine with that.